source/Ship.cpp

```diff
@@ -3946,12 +3946,39 @@ double Ship::BestFuel(const string &type, const string &subtype, double defaultF
 		double jumpRange = baseAttributes.Get("jump range");
 		if(!jumpRange)
 			jumpRange = System::DEFAULT_NEIGHBOR_DISTANCE;
+
+		double fuel = baseAttributes.Get("jump fuel");
+
+		double mass = Mass();
+		double driveMassExp = 0;
+		double driveMassRef = 400;
+		double driveDistExp = 0;
+		double driveDistRef = 100;
+
+		if((baseAttributes.Get("drive mass reference") || baseAttributes.Get("drive mass exponent")) && mass > 0)
+		{
+			driveMassRef = max(baseAttributes.Get("drive mass reference"), .01);
+			driveMassExp = max(baseAttributes.Get("drive mass exponent"), 0.);
+			fuel *= pow((mass / driveMassRef), driveMassExp);
+		}
+		if((baseAttributes.Get("drive distance reference") || baseAttributes.Get("drive distance exponent")) && jumpDistance > 0)
+		{
+			driveDistRef = max(baseAttributes.Get("drive distance reference"), .01);
+			driveDistExp = max(baseAttributes.Get("drive distance exponent"), 0.);
+			fuel *= pow((jumpDistance / driveDistRef), driveDistExp);
+		}
+
+		// if a "startup" fuel is provided, add that to the above formula.
+		// It's a constant fuel consumption regardless of mass or distance of jump.
+		if(baseAttributes.Get("jump startup fuel") > 0)
+			fuel += baseAttributes.Get("jump startup fuel");
```

Comment:
```suggestion
		fuel += baseAttributes.Get("jump startup fuel");
```
I think its best to do it this way, since jump startup fuel cannot be negative. If you want to keep on doing it your way though, at least store it in a variable instead of querying it twice.